### PR TITLE
Fix CLRDataCreateInstance export. This was breaking SOS.

### PR DESCRIPTION
**Pull request by mikem8361**
Added the right cmake magic so the function is exported properly.

### CODE REVIEW

File: src/dlls/mscordac/mscordac.cpp [hunk 1]
```diff
@@ -1,21 +1,3 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
```
Comment (hoyosjs) [inline src/dlls/mscordac/mscordac.cpp]: Is there any reason to leave an empty file?

Comment (mikem8361) [reply src/dlls/mscordac/mscordac.cpp]: I figured it needs a file to build the .so/.dll but I don't know for sure.

**Comment by mikem8361** [comment]
@hoyosjs have you seen thesefailures (in the Jenkin legs) in my PR before?

File: src/dlls/mscordac/mscordac.cpp (hunk 1 above)
Comment (hoyosjs) [reply src/dlls/mscordac/mscordac.cpp]: Can we leave a comment then so that cleanups don't delete it?

**Comment by hoyosjs** [comment]
No, can't say I've seen those. OSx has been failing fairly often lately, but hadn't seen a failure for CentOS like this one, and I don't remember seeing `tree->gtType == TYP_STRUCT` before in any PR recently as the assertion suggest, or how it would be cause by your change.

File: src/debug/daccess/daccess.cpp [hunk 2]
```diff
@@ -40,9 +40,7 @@ HINSTANCE g_thisModule;
 
 extern VOID STDMETHODCALLTYPE TLS_FreeMasterSlotIndex();
 
-#if FEATURE_PAL
-DLLEXPORT // For Win32 PAL LoadLibrary emulation
-#endif
+DLLEXPORT
```
Comment (jkotas) [inline src/debug/daccess/daccess.cpp]: DLLMain needs to be exported for PAL only. This ifdef was here to make that clear.

Comment (mikem8361) [reply src/debug/daccess/daccess.cpp]: Windows doesn't need DLLMain?

Comment (mikem8361) [reply src/debug/daccess/daccess.cpp]: It isn't needed on Windows because .def exports it.

Comment (mikem8361) [reply src/debug/daccess/daccess.cpp]: the .def file

Comment (jkotas) [reply src/debug/daccess/daccess.cpp]: DllMain does not need to be exported on Windows. It needs to be exported for PAL only to make DllMain emulation work.

Discussed in https://github.com/dotnet/coreclr/pull/21924#issuecomment-462067477

Comment (mikem8361) [reply src/debug/daccess/daccess.cpp]: Other than that is the PR ok?

**Pull request merged**